back/engines/free/user_confirmation/app/interactors/user_confirmation/confirm_user.rb

```diff
@@ -17,8 +18,14 @@ def validate_user
       fail_with_error! :user, :blank
     end
 
+    def validate_code_sent
+      return if user.email_confirmation_code_sent_at
+
+      fail_with_error! :code, :not_sent
+    end
```

Comment:
@kogre Added this check for `email_confirmation_code_sent_at` during code confirmation. This would prevent users from confirming their accounts if we have no record of the confirmation email being sent.

This would be a way of knowing if things are going wrong, but blocking users from confirming their accounts due to a bug sounds like a bad idea.